File: esports-backend/src/main/java/com/esports/entity/Service.java
package com.esports.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "services")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Service {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "freelancer_id", nullable = false)
    @JsonIgnore
    private User freelancer;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Double price;

    @Column(name = "delivery_time")
    private Integer deliveryTime;

    private String category;

    @Column(name = "is_active")
    private Boolean active = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @Column(name = "photo1_url", columnDefinition = "TEXT")
    private String photo1Url;

    @Column(name = "photo2_url", columnDefinition = "TEXT")
    private String photo2Url;

    @Column(name = "photo3_url", columnDefinition = "TEXT")
    private String photo3Url;

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: esports-backend-dotnet/Upkart1/Entities/Service.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Upkart1.Entities
{
    [Table("services")]
    public class Service
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        [Column("freelancer_id")]
        public long FreelancerId { get; set; }
        public User Freelancer { get; set; } = null!;

        [Required]
        [Column("title")]
        public string Title { get; set; } = null!;

        [Column("description", TypeName = "nvarchar(max)")]
        public string? Description { get; set; }

        [Required]
        [Column("price")]
        public double Price { get; set; }

        [Column("delivery_time")]
        public int? DeliveryTime { get; set; }

        [Column("category")]
        public string? Category { get; set; }

        [Column("is_active")]
        public bool Active { get; set; } = true;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [Column("photo1_url", TypeName = "nvarchar(max)")]
        public string? Photo1Url { get; set; }

        [Column("photo2_url", TypeName = "nvarchar(max)")]
        public string? Photo2Url { get; set; }

        [Column("photo3_url", TypeName = "nvarchar(max)")]
        public string? Photo3Url { get; set; }

        public ICollection<Project> Projects { get; set; } = new List<Project>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
    }
}
